packages/block-library/src/form-input/block.json

```diff
@@ -0,0 +1,61 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "core/form-input",
+	"title": "Input field",
+	"category": "common",
+	"parent": [ "core/form" ],
+	"description": "The basic building block for forms.",
+	"keywords": [ "input", "form" ],
+	"textdomain": "default",
+	"icon": "forms",
+	"attributes": {
+		"type": {
+			"type": "string",
+			"default": "text"
+		},
+		"name": {
+			"type": "string"
+		},
+		"label": {
+			"type": "string",
+			"default": "Label",
+			"selector": ".wp-block-form-input__label-content",
+			"source": "html",
+			"__experimentalRole": "content"
+		},
+		"inlineLabel": {
+			"type": "boolean",
+			"default": false
+		},
+		"required": {
+			"type": "boolean",
+			"default": false,
+			"selector": ".wp-block-form-input__input",
+			"source": "attribute",
+			"attribute": "required"
+		},
+		"placeholder": {
+			"type": "string",
+			"selector": ".wp-block-form-input__input",
+			"source": "attribute",
+			"attribute": "placeholder",
+			"__experimentalRole": "content"
+		}
```

Comment:
It would be nice if there was a way to also easily define additional properties such as `min`, `max`, `pattern` etc. I don't think there should be a UI for it currently but if a pattern wanted to define them or you went into the code editor to add them that shouldn't cause validation errors.

Reply:
Of definitely, there are many additional attributes we can add without a UI! 
For the first version of this however, I wanted to keep things clean and make it easier to review and merge. 
If we merge the initial (rather basic) implementation, we can do followup PRs to add additional features and enrich it 👍
Unless you feel this should be part of the initial PR?